monoio-http-client/src/client/mod.rs

```diff
@@ -70,147 +75,101 @@ impl Builder {
         Self::default()
     }
 
-    pub fn http1_client(&mut self) -> &mut Self {
+    pub fn http1_client(mut self) -> Self {
         self.connection_config.proto = Proto::Http1;
         self
     }
 
-    pub fn http2_client(&mut self) -> &mut Self {
+    pub fn http2_client(mut self) -> Self {
         self.connection_config.proto = Proto::Http2;
         self
     }
 
-    pub fn http_auto(&mut self) -> &mut Self {
+    pub fn http_auto(mut self) -> Self {
         self.connection_config.proto = Proto::Auto;
         self
     }
 
-    pub fn max_idle_connections(&mut self, conns: usize) -> &mut Self {
+    pub fn max_idle_connections(mut self, conns: usize) -> Self {
         self.global_config.max_idle_connections = conns;
         self
     }
 
-    pub fn http2_max_frame_size(&mut self, sz: u32) -> &mut Self {
+    pub fn http2_max_frame_size(mut self, sz: u32) -> Self {
         self.connection_config.h2_builder.max_frame_size(sz);
         self
     }
 
-    pub fn http2_max_send_buf_size(&mut self, sz: usize) -> &mut Self {
+    pub fn http2_max_send_buf_size(mut self, sz: usize) -> Self {
         self.connection_config.h2_builder.max_send_buffer_size(sz);
         self
     }
 
-    pub fn http2_max_concurrent_reset_streams(&mut self, max: usize) -> &mut Self {
+    pub fn http2_max_concurrent_reset_streams(mut self, max: usize) -> Self {
         self.connection_config.h2_builder.max_send_buffer_size(max);
         self
     }
 
-    pub fn http2_initial_stream_window_size(&mut self, size: u32) -> &mut Self {
+    pub fn http2_initial_stream_window_size(mut self, size: u32) -> Self {
         self.connection_config.h2_builder.initial_window_size(size);
         self
     }
 
-    pub fn http2_initial_connection_window_size(&mut self, size: u32) -> &mut Self {
+    pub fn http2_initial_connection_window_size(mut self, size: u32) -> Self {
         self.connection_config
             .h2_builder
             .initial_connection_window_size(size);
         self
     }
 
-    pub fn http2_max_concurrent_streams(&mut self, max: u32) -> &mut Self {
+    pub fn http2_max_concurrent_streams(mut self, max: u32) -> Self {
         self.connection_config
             .h2_builder
             .max_concurrent_streams(max);
         self
     }
 
-    pub fn build_http1(self) -> Client {
+    pub fn build(self) -> Client {
         Client::new(self.global_config, self.connection_config)
     }
 
-    pub fn build_http2(mut self) -> Client {
-        self.http2_client();
-        Client::new(self.global_config, self.connection_config)
-    }
-
-    pub fn build_auto(mut self) -> Client {
```

Comment:
Can we keep the build_* functions?  `build_http1` can be used for HTTP1 **only** client and same for http2. `build_auto` is for the Auto mode, where we support both HTTP1 and HTTP2, we use the version header in the request to decide which type to use. The default implementation for Client used `build_auto`

Reply:
I think it could be done with `.http_auto().build()`?